### ISSUE TITLE
Did someone succeed to run AdaGram on an Ubuntu server?

### ISSUE THREAD

**Issue by ClaudeCoulombe** [sbos/AdaGram.jl#27]
Greetings folks,

I've got a dirty memory access error (bus error) when running the Julia code loading a large AdaGram model `vm, dict = AdaGram.load_model(MODEL_PATH)` The kind of error that no one wants to have ... But everything works well on my Macbook?

There could be many causes: First, Julia installation (since for performance, Julia is very close to C) and I've used Anaconda, the compiler (Clang), the OS, the CPU and possibly other factors. Then, I've checked, there is enough RAM memory.

But first, I would like to know if someone has succeeded to run AdaGram on a Linux Ubuntu server?

---------

`
signal (7): Bus error
while loading /home/ccoulombe/AdaGram.jl/disambiguate.jl, in expression starting on line 22
macro expansion at ./cartesian.jl:62 [inlined]
macro expansion at ./multidimensional.jl:431 [inlined]
_unsafe_batchsetindex! at ./multidimensional.jl:423
_setindex! at ./multidimensional.jl:372 [inlined]
setindex! at ./abstractarray.jl:840 [inlined]
#9 at /home/ccoulombe/anaconda3/envs/my_env/share/julia/site/v0.5/AdaGram/src/AdaGram.jl:64
#620 at ./multi.jl:1030
run_work_thunk at ./multi.jl:1001
run_work_thunk at ./multi.jl:1010 [inlined]
#617 at ./event.jl:68
unknown function (ip: 0x7f4b5c48b89f)
warning: parsing line table prologue at 0x00003c27 should have ended at 0x00003ffc but it ended at 0x00003ce4
jl_call_method_internal at (null):0 [inlined]
jl_apply_generic at (null):1950
warning: parsing line table prologue at 0x0000e1c8 should have ended at 0x0000e52f but it ended at 0x0000e285
jl_apply at (null):0 [inlined]
start_task at (null):254
unknown function (ip: 0xffffffffffffffff)
Allocations: 9998583 (Pool: 9997420; Big: 1163); GC: 17
./run.sh: line 8:  2712 Bus error               (core dumped) LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$DIR/lib julia "$@"
`

**Comment by lopuhin** [comment]
@ClaudeCoulombe I did run it successfully on Ubuntu server, and I also remember getting similar errors when training big models, and I thought that in my case the reason was not enough memory (so I reduced dictionary size). On OS X you can usually allocate much more memory before running into allocation errors than on Ubuntu, I think. But it could be something completely different, just my hypothesis.

**Comment by ClaudeCoulombe** [comment]
@lopuhin Nice to know! I've already install 7 GB of RAM on the server and my estimation of the requirements are just over 4 GB but I will check it twice! You're feedback is very helpful.

**Comment by ClaudeCoulombe** [comment]
Finally, I've succeeded to run AdaGram with a large model (1.5 GB) on a Linux Ubuntu server with 16 GB of RAM, but not with the 3 GB model I've tried at the beginning. However,  this 3 GB model is still running well on my MacBook Pro (16 GB of RAM). 

I suspect maybe a problem with the alignment of the model complex structure read from file and put in RAM memory and the Julia / C functions which access to it. But I'm not sure...

Note: I've installed Julia and Python manually (not using Anaconda)
